types/react-redux/index.d.ts

```diff
@@ -316,18 +319,17 @@ export interface ConnectOptions {
     withRef?: boolean
 }
 
-export interface ProviderProps {
+export interface ProviderProps<A extends Action> {
```

Comment:
this still needs a default to avoid breaking people. Even though JSX invocations of `Provider` can infer `A` from `store`, type annotations would need to change from `ProviderProps` to `ProviderProps<SomeAction>`.